### PR TITLE
Add ormar support

### PR DESCRIPTION
**Pull request by collerek**
Hi,

I'm creator of `ormar` - async mini orm with `pydantic` validation and `fastapi` in mind.
Since I saw you support a lot of backends but not `ormar` I decided to add support for it.

`ormar` is very similiar to `encode/orm` yet has much more features and can be used directly with fastapi as response and request params.

Hope you don't mind and let me know what you think!

### CODE REVIEW

File: fastapi_pagination/ext/ormar.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from typing import Optional, Type, Union
+
+from ormar import Model, QuerySet
+
+from ..api import create_page, resolve_params
+from ..bases import AbstractPage, AbstractParams
+
+
+async def paginate(query: Union[QuerySet, Type[Model]], params: Optional[AbstractParams] = None) -> AbstractPage:
+    if not isinstance(query, QuerySet):
```
Comment (uriyyo) [inline fastapi_pagination/ext/ormar.py:10]: Please, add tests to cover case when query is a model instance.

Example for `gino`:
https://github.com/uriyyo/fastapi-pagination/blob/25bcbb9fe101886f0481472bf352876f9c9e6517/tests/ext/test_gino.py#L35-L44

File: pyproject.toml [hunk 2]
```diff
@@ -32,6 +32,7 @@ databases = { version = "^0.4.0", extras = ["postgresql", "sqlite", "mysql"], op
 orm = { version = "^0.1.5", optional = true }
 tortoise-orm = { version = "^0.16.18", extras = ["asyncpg", "aiomysql", "aiosqlite"], optional = true }
 asyncpg = { version = "0.22.0", optional = true }
+ormar = "^0.9.8"
```
Comment (uriyyo) [inline pyproject.toml:35]: Please make `ormar` optional dependency and add `extras` to install `fastapi-pagination` with `ormar` integration.

File: tests/ext/test_ormar.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+def User(db, metadata):
+    # weird syntax cause otherwise class definition is evaluated
+    # before fixtures are resolved, and ormar uses Metaclass that already
+    # requires sqlalchemy.Metadata() to exist
+    definition = {
+        "Meta": type("Meta", (ModelMeta,), dict(database=db, metadata=metadata)),
+        "id": Integer(primary_key=True),
+        "name": String(max_length=100),
+    }
+    User = type("User", (Model,), definition)
```
Comment (uriyyo) [inline tests/ext/test_ormar.py:35]: Could we write smth like this?
```py
@fixture(scope="session")
def User(db, metadata):
    class User(Model):
        class Meta:
            tablename = "users"
            database = db
            metadata = metadata

        id = Integer(primary_key=True)
        name = String(max_length=100)
    
    return User
```

**Review by uriyyo — CHANGES_REQUESTED** [review]
LGTM in general.

Actually I was planing to add `ormar` integration🙂

I love the way how `ormar` use `pydantic`, it's really awesome 💫

Thanks for making this PR, I am really appreciate it.

File: pyproject.toml (hunk 2 above)
Comment (collerek) [reply pyproject.toml:35]: This one I wanted to discuss as I don't use poetry :)
Ormar have it's own extras:
```
extras_require={
        "postgresql": ["asyncpg", "psycopg2"],
        "mysql": ["aiomysql", "pymysql"],
        "sqlite": ["aiosqlite"],
        "orjson": ["orjson"],
        "crypto": ["cryptography"]
    }
```
The sync versions are used in sqlalchemy to create tables etc. Which one should I include? And should it be my name (i.e. crypto) or package name? How does it work here?

Comment (uriyyo) [reply pyproject.toml:35]: This should look like this:
```toml
[tool.poetry.dependencies]
ormar = { version = "^0.9.8", optional = true}

[tool.poetry.extras]
ormar = ["ormar"]
```

Also, don't forget to update `all` extras)

Comment (collerek) [reply pyproject.toml:35]: OK but what about ormar extras?
Like tortoise has it's own?
`tortoise-orm = { version = "^0.16.18", extras = ["asyncpg", "aiomysql", "aiosqlite"]...`

Comment (uriyyo) [reply pyproject.toml:35]: My bad, `tortoise-orm` extras should be removed

File: tests/ext/test_ormar.py (hunk 3 above)
Comment (collerek) [reply tests/ext/test_ormar.py:35]: Yeah I was wondering what the hell is going on until just recently I realized that line `metadata=metadata` was redeclaring the metadata so had to change the fixture name and it's all good. :D

**Comment by uriyyo** [comment]
@collerek Please run `poetry lock` to update lock file.

**Comment by collerek** [comment]
@uriyyo yeah just realized - time to learn the basics of poetry :)
pipenv, pip+virtualenv, poetry - we have too much too chose from :)

**Comment by uriyyo** [comment]
> pipenv, pip+virtualenv, poetry - we have too much too chose from :)

Totally agree with you. Will be great to have one tool to manage dependencies)

**Review by uriyyo — APPROVED** [review]
LGTM💫

**Pull request merged**